File: python/剑指offer/面试题10- II. 青蛙跳台阶问题.py
class Solution:
    def numWays(self, n: int) -> int:
        result = [1]

        for i in range(1,n+1):
            if i == 1:
                result.append(1)
            elif i == 2:
                result.append(2)
            else:
                result.append(result[i-1]+result[i-2])
        return result[n] % 1000000007

s = Solution()
n = 7
print(s.numWays(n))

File: go/剑指offer/面试题10- II. 青蛙跳台阶问题.go
package main

func numWays(n int) int {
	if n == 0 {
		return 1
	}
	if n <= 2 {
		return n
	}
	a, b := 1, 2
	for i := 3; i < n+1; i++ {
		a, b = b, (a+b)%1000000007
	}
	return b
}
